BeagleBone_black/src/communications/communications.cpp

```diff
@@ -67,29 +69,42 @@ Communications::~Communications()
 int Communications::sendData(std::string message)
 {
   // Incoming strings should be terminated by "...\n".
-  memset(buffer, '\0', 256);
+  memset(buffer_, '\0', 256);
   const char *data = message.c_str();
-  n_ = write(sockfd_, data, message.length());  // ‘_size_t write(int, const void*, size_t)’
+  int n = write(sockfd_, data, message.length());  // ‘_size_t write(int, const void*, size_t)’
 
-  if (n_ < 0) {
+  if (n < 0) {
     log_.ERR("COMN", "CANNOT READ FROM SOCKET.\n");
   }
 
-  return atoi(buffer);
+  return atoi(buffer_);
+}
+
+int Communications::receiveRunLength()
+{
+  int n = read(sockfd_, buffer_, 255);
```

Comment:
could throw an CANNOT READ FROM SOCKET error if n < 0 here

Reply:
What does that mean? Would the program crash?

Reply:
The program wouldn't crash at that point. It would crash a millisecond later when the client tried to send its next block of data to the server. All this really does is show us that connection was first lost at that point when the client tried to receive the new track length. To be honest it doesn't make a significant difference if we add or leave that.